modules/indicia_setup/db/version_6_1_0/202107151745_list_text_value_of_float_attribute.sql

```diff
@@ -0,0 +1,590 @@
+-- For each view, `list_*_attribute values`, alter the `value` field
+-- for float data types to return the `text_value`, if there is one.
+-- The text value is the entered value, including trailing zeroes,
+-- which we need when a decimal validation rule is in place checking
+-- the number of digits after the decimal place.
+-- Decimal validation and ranges do not play nicely together.
+
+-- I.e.
+-- 
+-- WHEN 'F'::bpchar THEN av.float_value::text ||
+-- CASE
+--     WHEN (a.data_type = ANY (ARRAY['I'::bpchar, 'F'::bpchar])) AND a.allow_ranges = true THEN COALESCE(' - '::text || av.upper_value::text, ''::text)
+--     ELSE ''::text
+-- END
+--
+-- becomes
+--
+-- WHEN 'F'::bpchar THEN
+-- CASE
+--     WHEN av.text_value IS NOT NULL THEN av.text_value
+--     ELSE av.float_value::text ||
+--     CASE
+--         WHEN (a.data_type = ANY (ARRAY['I'::bpchar, 'F'::bpchar])) AND a.allow_ranges = true THEN COALESCE(' - '::text || av.upper_value::text, ''::text)
+--         ELSE ''::text
+--     END
+-- END
+
+
+CREATE OR REPLACE VIEW indicia.list_location_attribute_values
```

Comment:
The schema prefix "indicia." should be removed throughout this script.

Reply:
Done